### PR TITLE
Alethe: Add ARITH_TRICHOTOMY to updater

### PR DESCRIPTION
**Pull request by Lachnitt**
Translation of the ARITH_TRICHOTOMY rule into the Alethe calculus.

Co-authored-by: Haniel Barbosa hanielbbarbosa@gmail.com

### CODE REVIEW

File: src/proof/alethe/alethe_post_processor.cpp [hunk 1]
```diff
@@ -1384,6 +1384,250 @@ bool AletheProofPostprocessCallback::update(Node res,
...
+    //
+    // * the corresponding proof node is C
+    case PfRule::ARITH_TRICHOTOMY:
+    {
+      bool success = true;
+      Node equal;
+      Node lesser;
+      Node greater;
+
+      if (res.getKind() == kind::EQUAL)
```
Comment (HanielB) [inline src/proof/alethe/alethe_post_processor.cpp:1415]:
```suggestion
      Kind k = res.getKind();
      if (k == kind::EQUAL)
```
Also change occurrences below.

File: src/proof/alethe/alethe_post_processor.cpp [hunk 2]
```diff
@@ -1384,6 +1384,250 @@ bool AletheProofPostprocessCallback::update(Node res,
...
+        x = equal[0][0];
+        c = equal[0][1];
+      }
+      else
+      {
+        x = equal[0];
+        c = equal[1];
+      }
+      Node vp_child1 = children[0];
+      Node vp_child2 = children[1];
```
Comment (HanielB) [inline src/proof/alethe/alethe_post_processor.cpp:1472]:
```suggestion
      Node vp_child1 = children[0], vp_child2 = children[1];
```

Comment (HanielB) [inline src/proof/alethe/alethe_post_processor.cpp:1472]:
```suggestion
      Node vp_child1 = children[0], vp_child2 = children[1];
```

File: src/proof/alethe/alethe_post_processor.cpp [hunk 3]
```diff
@@ -1384,6 +1384,250 @@ bool AletheProofPostprocessCallback::update(Node res,
...
+
+      // Preprocessing
+      if (res == equal || res == greater)
+      {  // C = (= x c) or C = (> x c)
+        // lesser = (>= x c)
+        Node vpc2 = nm->mkNode(kind::SEXPR,
+                               d_cl,
+                               nm->mkNode(kind::EQUAL,
+                                          nm->mkNode(kind::GEQ, x, c),
+                                          nm->mkNode(kind::LEQ, c, x)));
```
Comment (HanielB) [inline src/proof/alethe/alethe_post_processor.cpp:1482]:
```suggestion
        Node vpc2 = nm->mkNode(
            kind::SEXPR,
            d_cl,
            nm->mkNode(kind::GEQ, x, c).eqNode(nm->mkNode(kind::LEQ, c, x)));
```

File: src/proof/alethe/alethe_post_processor.cpp [hunk 4]
```diff
@@ -1384,6 +1384,250 @@ bool AletheProofPostprocessCallback::update(Node res,
...
+        // greater = (<= x c) or greater = (not (= x c)) -> no preprocessing
+        // necessary
+        if (res == equal)
+        {
+          vp_child2 = greater;
+        }
+        else
+        {
+          vp_child2 = equal;
+        }
```
Comment (HanielB) [inline src/proof/alethe/alethe_post_processor.cpp:1512]:
```suggestion
        vp_child2 = res == equal ? greater : equal;
```

**Review by HanielB — CHANGES_REQUESTED** [review]
Looks good, just some style requests

File: src/proof/alethe/alethe_post_processor.cpp [hunk 5]
```diff
@@ -1384,6 +1384,250 @@ bool AletheProofPostprocessCallback::update(Node res,
...
+      {  // no postprocessing necessary
+        return success
+               && addAletheStep(AletheRule::RESOLUTION,
+                                res,
+                                nm->mkNode(kind::SEXPR, d_cl, res),
+                                {vp2, vp_child1, vp_child2},
+                                {},
+                                *cdp);
+      }
+      else if (res == greater)
```
[the remark this answers is not in the document]
Comment (Lachnitt) [reply src/proof/alethe/alethe_post_processor.cpp:1533]: Why this?

[the remark this answers is not in the document]
Comment (HanielB) [reply src/proof/alethe/alethe_post_processor.cpp:1533]: Because it's after a `return`.

**Review by HanielB — APPROVED** [review]

**Pull request merged**